packages/tools/viewer/src/viewer.ts

```diff
@@ -1501,6 +1816,89 @@ export class Viewer implements IDisposable {
         return true;
     }
 
+    /**
+     * Get hotspot world and screen values from a named hotspot
+     * @param name slot of the hot spot
+     * @param result resulting world and screen positions
+     * @returns world position, world normal and screen space coordinates
+     */
+    public queryHotSpot(name: string, result: ViewerHotSpotResult): boolean {
+        return this._queryHotSpot(name, result) != null;
+    }
+
+    /**
+     * Updates the camera to focus on a named hotspot.
+     * @param name The name of the hotspot to focus on.
+     * @returns true if the hotspot was found and the camera was updated, false otherwise.
+     */
+    public focusHotSpot(name: string): boolean {
+        const result = new ViewerHotSpotResult();
+        const query = this._queryHotSpot(name, result);
+        if (query) {
+            this.pauseAnimation();
+            const cameraOrbit = query.cameraOrbit ?? [undefined, undefined, undefined];
+            this._camera.interpolateTo(cameraOrbit[0], cameraOrbit[1], cameraOrbit[2], new Vector3(result.worldPosition[0], result.worldPosition[1], result.worldPosition[2]));
+            return true;
+        }
+        return false;
+    }
+
+    private _queryHotSpot(name: string, result: ViewerHotSpotResult): Nullable<HotSpot> {
+        const hotSpot = this.hotSpots?.[name];
+        if (hotSpot) {
+            if (this.getHotSpotToRef(hotSpot, result)) {
+                return hotSpot;
+            }
+        }
+        return null;
+    }
+
+    private async _addCameraHotSpot(camera: Camera, signal?: AbortSignal) {
+        if (camera !== this._camera) {
+            const hotSpot = await this._createHotSpotFromCamera(camera);
+            if (hotSpot && !signal?.aborted) {
+                this.hotSpots = {
+                    ...this.hotSpots,
+                    [`camera-${camera.name}`]: hotSpot,
+                };
+            }
+        }
+    }
+
+    private _removeCameraHotSpot(camera: Camera) {
+        delete this.hotSpots[`camera-${camera.name}`];
+        this.hotSpots = { ...this.hotSpots };
+    }
+
+    private _toggleCamerasAsHotSpots() {
+        if (!this.camerasAsHotSpots) {
+            this._camerasAsHotSpotsAbortController?.abort();
+            this._camerasAsHotSpotsAbortController = null;
+            this._scene.cameras.forEach((camera) => this._removeCameraHotSpot(camera));
+        } else {
+            const abortController = (this._camerasAsHotSpotsAbortController = new AbortController());
+            this._scene.cameras.forEach((camera) => this._addCameraHotSpot(camera, abortController.signal));
+        }
+    }
+
+    /**
+     * Creates a world HotSpot from a camera.
+     * @param camera The camera to create a HotSpot from.
+     * @returns A HotSpot created from the camera.
+     */
+    private async _createHotSpotFromCamera(camera: Camera): Promise<Nullable<HotSpot>> {

```

Comment:
```suggestion
    private async _createHotSpotFromCameraAsync(camera: Camera): Promise<Nullable<HotSpot>> {
```

Reply:
Let's follow up after this PR to discuss naming conventions and whether we want to deprecate and rename.